app/(gcforms)/[locale]/(form administration)/form-builder/[id]/responses/[statusFilter]/components/ManageFormAccessDialog/ManageFormAccessDialog.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import { Button } from "@clientComponents/globals";
+import { Dialog, useDialogRef } from "@formBuilder/components/shared";
+import { useCustomEvent } from "@lib/hooks/useCustomEvent";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { InviteUsers } from "./InviteUsers";
+import { ManageUsers } from "./ManageUsers";
+import { ManageFormAccessDialogContext } from "./ManageFormAccessDialogContext";
+import { isValidGovEmail } from "@lib/validation/validation";
+import { useTranslation } from "@i18n/client";
+
+type ManageFormAccessDialogProps = {
+  formId: string;
+};
+
+export const ManageFormAccessDialog = ({ formId }: ManageFormAccessDialogProps) => {
+  const { t } = useTranslation("manage-form-access");
+
+  const manageFormAccessDialogContext = useContext(ManageFormAccessDialogContext);
+
+  if (!manageFormAccessDialogContext) {
+    throw new Error("ManageFormAccessDialog must be used within a ManageFormAccessDialogProvider");
+  }
+
+  const { isOpen, setIsOpen, setFormId, emailList, setEmailList } = manageFormAccessDialogContext;
+
+  const dialogRef = useDialogRef();
+  const { Event } = useCustomEvent();
+  const [isInvitationScreen, setIsInvitationScreen] = useState(false);
+  const isManagementScreen = !isInvitationScreen;
+
+  /**
+   * Open the dialog
+   */
+  const handleOpenDialog = useCallback(() => {
+    setIsInvitationScreen(false);
+    setEmailList([]);
+    setIsOpen(true);
+  }, [setEmailList, setIsOpen]);
+
+  /**
+   * Close the dialog and reset the data
+   */
+  const handleCloseDialog = () => {
+    setIsOpen(false);
+    setIsInvitationScreen(false);
+  };
+
+  useEffect(() => {
+    setFormId(formId);
```

Comment:
If this is coming in from a prop I think we can just remove this no?

Reply:
This is pushing the state up to the context so other components that need it can get it.